File: temp_replace_methodology.js
const fs = require('fs');
const path = require('path');
const filePath = path.join(__dirname, 'src', 'composants', 'Portfolio.jsx');
const text = fs.readFileSync(filePath, 'utf8');
const lines = text.split(/\r?\n/);
const startMarker = "          <div style={{ padding: isMobile ? '18px 16px' : '22px 26px', display: 'flex', flexDirection: 'column', gap: 16 }}>",
      endMarker = "        </div>";
const startIdx = lines.findIndex(l => l.trim() === startMarker.trim());
const styleIdx = lines.findIndex(l => l.includes("<style>{`@keyframes mblink {"));
if (startIdx === -1 || styleIdx === -1) {
  console.error('Markers not found', { startIdx, styleIdx });
  process.exit(1);
}
let endIdx = -1;
for (let i = styleIdx - 1; i > startIdx; i -= 1) {
  if (lines[i].trim() === endMarker.trim()) {
    endIdx = i;
    break;
  }
}
if (endIdx === -1) {
  console.error('End divider not found', { startIdx, styleIdx });
  process.exit(1);
}
const replacementLines = [
"          <div style={{",
"            display: 'flex',",
"            flexDirection: isMobile ? 'column' : 'row',",
"            gap: 6,",
"            height: isMobile ? 'auto' : 340,",
"          }}>",
"            {etapes.map((e, i) => {",
"              const isActive = i === etapeActive",
"              return (",
"                <div",
"                  key={e.n || i}",
"                  role=\"button\"",
"                  tabIndex={0}",
"                  aria-label={e.titre}",
"                  onClick={() => setEtapeActive(i)}",
"                  onKeyDown={(ev) => { if (ev.key === 'Enter' || ev.key === ' ') setEtapeActive(i) }}",
"                  style={{",
"                    position: 'relative',",
"                    flex: isMobile ? 'none' : (isActive ? 5 : 1),",
"                    minWidth: 0,",
"                    height: isMobile ? (isActive ? 180 : 56) : '100%',",
"                    cursor: 'pointer',",
"                    borderRadius: 14,",
"                    border: `1px solid ${isActive ? eff.borderStrong : eff.borderMedium}`",
"                    borderLeft: isActive ? `2px solid ${a}` : `1px solid ${eff.borderMedium}`",
"                    background: isActive ? eff.cardBg : `rgba(${aRgb},0.02)`",
"                    overflow: 'hidden',",
"                    transition: isMobile",
"                      ? 'height 0.4s cubic-bezier(0.22,1,0.36,1), background 0.3s'",
"                      : 'flex 0.4s cubic-bezier(0.22,1,0.36,1), background 0.3s',",
"                    display: 'flex',",
"                  }}",
"                >",
"                  <span style={{",
"                    position: 'absolute',",
"                    fontFamily: 'Fraunces, serif',",
"                    fontWeight: 800,",
"                    color: a,",
"                    lineHeight: 1,",
"                    userSelect: 'none',",
"                    pointerEvents: 'none',",
"                    ...(isActive",
"                      ? { fontSize: isMobile ? 90 : 190, right: -10, bottom: -30, opacity: 0.10 }",
"                      : isMobile",
"                        ? { fontSize: 20, left: 18, top: '50%', transform: 'translateY(-50%)', opacity: 0.5 }",
"                        : { fontSize: 26, top: 14, left: '50%', transform: 'translateX(-50%)', opacity: 0.35 }",
"                    ),",
"                  }}>",
"                    {e.n || String(i + 1).padStart(2, '0')}",
"                  </span>",
"",
"                  {isActive ? (",
"                    <div style={{",
"                      position: 'relative',",
"                      padding: isMobile ? '20px 20px 18px' : '28px 26px',",
"                      display: 'flex',",
"                      flexDirection: 'column',",
"                      justifyContent: 'flex-end',",
"                      width: '100%',",
"                    }}>",
"                      <span style={{ fontFamily: 'Space Mono, monospace', fontSize: 11, color: a, marginBottom: 8, letterSpacing: '0.1em' }}>",
"                        {e.n || String(i + 1).padStart(2, '0')}",
"                      </span>",
"                      <span style={{ fontFamily: 'Fraunces, serif', fontWeight: 700, fontSize: isMobile ? 18 : 21, color: eff.textPrimary, marginBottom: 8 }}>",
"                        {e.titre}",
"                      </span>",
"                      <span style={{ fontFamily: 'Inter, sans-serif', fontWeight: 300, fontSize: 13, color: eff.textMuted, lineHeight: 1.6, maxWidth: 300 }}>",
"                        {e.desc}",
"                      </span>",
"                    </div>",
"                  ) : isMobile ? (",
"                    <div style={{ display: 'flex', alignItems: 'center', width: '100%', padding: '0 18px 0 52px' }}>",
"                      <span style={{ fontFamily: 'Space Mono, monospace', fontSize: 12, color: eff.textMuted, letterSpacing: '0.05em' }}>",
"                        {e.titre}",
"                      </span>",
"                    </div>",
"                  ) : (",
"                    <span style={{",
"                      position: 'absolute',",
"                      bottom: 20,",
"                      left: '50%',",
"                      transform: 'translateX(-50%) rotate(180deg)',",
"                      writingMode: 'vertical-rl',",
"                      fontFamily: 'Space Mono, monospace',",
"                      fontSize: 11,",
"                      letterSpacing: '0.05em',",
"                      color: eff.textMuted,",
"                      whiteSpace: 'nowrap',",
"                    }}>",
"                      {e.titre}",
"                    </span>",
"                  )}",
"                </div>",
"              )",
"            })}",
"          </div>"
];
const newLines = lines.slice(0, startIdx).concat(replacementLines).concat(lines.slice(endIdx + 1));
fs.writeFileSync(filePath, newLines.join('\n'));
console.log('Replaced methodology block from', startIdx, 'to', endIdx);

File: temp_replace_methodology.py
from pathlib import Path

root = Path(__file__).resolve().parent
file_path = root / 'src' / 'composants' / 'Portfolio.jsx'
text = file_path.read_text(encoding='utf-8')
lines = text.splitlines()
start = next(i for i, line in enumerate(lines) if line.strip() == '<section id="pf-methodology" style={sectionStyle}>')
end = next(i for i, line in enumerate(lines) if line.strip() == '<section id="pf-contact" style={sectionStyle}>')
replacement = '''      <section id="pf-methodology" style={sectionStyle}>
        <div style={s.secNum}>05 // MÉTHODE</div>
        <h2 style={s.secTitle}>Ma <span style={s.accent}>Méthode</span> de Travail</h2>

        <div style={{
          background: eff.cardBg,
          border: `1px solid ${eff.borderStrong}`,
          borderRadius: 16,
          overflow: 'hidden',
          maxWidth: 820,
          margin: '0 auto',
          position: 'relative',
        }}>
          <div style={{
            position: 'absolute', top: 0, left: 40, right: 40, height: 1,
            background: `linear-gradient(90deg,transparent,rgba(${aRgb},0.3),transparent)`,
          }} />

          <div style={{
            display: 'flex',
            flexDirection: isMobile ? 'column' : 'row',
            gap: 10,
          }}>
            {etapes.map((e, i) => {
              const isActive = i === etapeActive
              return (
                <div
                  key={e.n || i}
                  role="button"
                  tabIndex={0}
                  aria-label={e.titre}
                  onClick={() => setEtapeActive(i)}
                  onKeyDown={(ev) => { if (ev.key === 'Enter' || ev.key === ' ') setEtapeActive(i) }}
                  style={{
                    position: 'relative',
                    flex: isMobile ? 'none' : isActive ? 3 : 1,
                    minWidth: 0,
                    height: isMobile ? (isActive ? 220 : 58) : 'auto',
                    cursor: 'pointer',
                    borderRadius: 14,
                    border: `1px solid ${isActive ? eff.borderStrong : eff.borderMedium}`,
                    background: isActive ? eff.cardBg : `rgba(${aRgb},0.04)`,
                    overflow: 'hidden',
                    transition: 'all 0.35s cubic-bezier(0.22,1,0.36,1)',
                    display: 'flex',
                    alignItems: 'stretch',
                  }}
                >
                  <span style={{
                    position: 'absolute',
                    color: a,
                    fontFamily: 'Fraunces, serif',
                    fontWeight: 800,
                    lineHeight: 1,
                    userSelect: 'none',
                    pointerEvents: 'none',
                    ...(isActive
                      ? { fontSize: isMobile ? 90 : 180, right: -10, bottom: -28, opacity: 0.10 }
                      : isMobile
                        ? { fontSize: 18, left: 16, top: '50%', transform: 'translateY(-50%)', opacity: 0.5 }
                        : { fontSize: 24, top: 16, left: '50%', transform: 'translateX(-50%)', opacity: 0.28 }
                    ),
                  }}>
                    {e.n || String(i + 1).padStart(2, '0')}
                  </span>

                  {isActive ? (
                    <div style={{
                      position: 'relative',
                      padding: isMobile ? '22px 20px 18px' : '30px 28px',
                      display: 'flex',
                      flexDirection: 'column',
                      justifyContent: 'flex-end',
                      width: '100%',
                    }}>
                      <span style={{
                        fontFamily: 'Space Mono, monospace',
                        fontSize: 11,
                        color: a,
                        letterSpacing: '0.18em',
                        marginBottom: 10,
                      }}>
                        {e.n || String(i + 1).padStart(2, '0')}
                      </span>
                      <span style={{
                        fontFamily: 'Fraunces, serif',
                        fontWeight: 700,
                        fontSize: isMobile ? 18 : 20,
                        color: eff.textPrimary,
                        marginBottom: 10,
                      }}>
                        {e.titre}
                      </span>
                      <span style={{
                        fontFamily: 'Inter, sans-serif',
                        fontWeight: 300,
                        fontSize: 13,
                        color: eff.textMuted,
                        lineHeight: 1.7,
                      }}>
                        {e.desc}
                      </span>
                    </div>
                  ) : (
                    <div style={{
                      display: 'flex',
                      alignItems: 'center',
                      justifyContent: isMobile ? 'flex-start' : 'center',
                      width: '100%',
                      padding: isMobile ? '0 20px 0 56px' : '22px 14px 22px',
                    }}>
                      <span style={{
                        fontFamily: 'Space Mono, monospace',
                        fontSize: isMobile ? 12 : 11,
                        color: eff.textMuted,
                        letterSpacing: '0.1em',
                        textAlign: isMobile ? 'left' : 'center',
                      }}>
                        {e.titre}
                      </span>
                    </div>
                  )}
                </div>
              )
            })}
          </div>
        </div>
      </section>'''

new_lines = lines[:start] + replacement.splitlines() + lines[end:]
file_path.write_text('\n'.join(new_lines) + '\n', encoding='utf-8')
print(f'Updated section from {start} to {end}')
